basic_module/distribution/src/lib.rs

```diff
@@ -0,0 +1,294 @@
+// Copyright 2020 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+mod core;
+mod error;
+mod impls;
+mod types;
+
+use coordinator::context::Context;
+use error::Error;
+use primitives::H256;
+use std::collections::{hash_map, BTreeMap, HashMap};
+use std::convert::{From, TryFrom};
+use std::mem;
+
+use ccrypto::blake256;
+use rlp::{Encodable, RlpStream};
+
+pub use ckey::Ed25519Public as Public;
+pub use ctypes::{BlockHash, BlockNumber, CommonParams, Header};
+use std::unimplemented;
+pub use types::BlockId;
+
+pub type StakeQuantity = u64;
+pub type Deposit = u64;
+pub type StorageId = u16;
+pub type StateResult<T> = Result<T, Error>;
+
+pub const CUSTOM_ACTION_HANDLER_ID: u64 = 2;
+
+#[derive(Clone, Debug)]
+pub struct IntermediateRewards {
+    current_reward: BTreeMap<Public, u64>,
+    calculated_reward: BTreeMap<Public, u64>,
+    total_min_fee: u64,
+    min_fee: u64,
+}
+
+impl IntermediateRewards {
```

Comment:
Where is the function loading the state from the storage?

Reply:
Ah, ok the function is not implemented for the struct but for a global function

Reply:
Yes, I implemented `get_intermediate_rewards()` as a global function.